distributed/tests/test_steal.py

```diff
@@ -1288,6 +1292,7 @@ async def test_reschedule_concurrent_requests_deadlock(c, s, *workers):
     assert msgs in (expect1, expect2, expect3)
 
 
+@pytest.mark.skip("executing heartbeats not considered yet")
 @gen_cluster(client=True, nthreads=[("127.0.0.1", 1)] * 3)
 async def test_correct_bad_time_estimate(c, s, *workers):
```

Comment:
This is the one functionality I couldn't restore so far. The problem is that upon every reevaluate_occupancy, we didn't only reevaluate the occupancy but if we detect a significant shift of occupancy, we'd recalculate the steal time ratio for all tasks in processing.
With this PR there is no place any more to reevaluate occupancy so this is no longer possible. More natural would be to recalculate tasks whenever a task group/prefix duration drifts but we'd need to track tasks of a taskgroup to make this work.
I'm currently not fully convinced that this is worth doing. Particularly since this only affects tasks with large network transfers and small occupancy. As it stands right now, this would only affects tasks with a transfer time to occupancy ratio of more than 257 which is typically only possible for lightning fast tasks anyhow.

Before engaging on this I would like to get https://github.com/dask/distributed/pull/7026 or a version of it done